Velaptor/UI/TextBox.cs

```diff
@@ -1,78 +1,371 @@
+
+                this.currentCharIndex = newIndex + 1;
+            }
+
+            textRight = this.charBounds.TextRight();
+            hasBounds = this.charBounds.Count > 0;
+            UpdateState(MutateType.PostMutate, hasBounds, textRight, KeyCode.Unknown, TextBoxEvent.DimensionalChange);
+
+            this.textCursor.Update();
+        }
+    }
+
+    /// <summary>
+    /// Gets the height of the <see cref="TextBox"/>.
+    /// </summary>
+    public override uint Height => this.height;
+
+    /// <summary>
+    /// Gets or sets the position of the <see cref="TextBox"/> on the screen.
+    /// </summary>
+    public override Point Position
+    {
+        get => base.Position;
+        set
+        {
+            if (base.Position == value)
+            {
+                return;
+            }
+
+            var movingLeft = value.X < base.Position.X;
+            var movingRight = value.X > base.Position.X;
+            var deltaX = Math.Abs(value.X - base.Position.X);
+            var deltaY = Math.Abs(value.Y - base.Position.Y);
+
+            var textRight = this.charBounds.TextRight();
+            var hasBounds = this.charBounds.Count > 0;
+
+            UpdateState(MutateType.PreMutate, hasBounds, textRight, KeyCode.Unknown, TextBoxEvent.Movement);
+
+            base.Position = value;
+
+            if (movingLeft)
+            {
+                this.charBounds.BumpAllToLeft(deltaX);
+                this.textAreaBounds.Position = this.textAreaBounds.Position with { X = this.textAreaBounds.Position.X - deltaX, };
+            }
+            else if (movingRight)
+            {
+                this.charBounds.BumpAllToRight(deltaX);
+                this.textAreaBounds.Position = this.textAreaBounds.Position with { X = this.textAreaBounds.Position.X + deltaX, };
+            }
+
+            UpdateBounds(KeyCode.Unknown);
+
+            if (deltaY > 0)
+            {
+                this.charBounds.BumpAllUp(deltaY);
+            }
+
+            UpdateTextPos();
+
+            this.textAreaBounds.Position = this.textAreaBounds.Position with { Y = Position.Y };
+
+            textRight = (int)(this.charBounds.IsEmpty() ? this.textAreaBounds.Right : this.charBounds.TextRight());
+            hasBounds = this.charBounds.Count > 0;
+            UpdateState(MutateType.PostMutate, hasBounds, textRight, KeyCode.Unknown, TextBoxEvent.Movement);
+            this.textCursor.Update();
+        }
+    }
 
     /// <summary>
     /// Loads the content of the <see cref="TextBox"/>.
     /// </summary>
-    public override void LoadContent() => this.text.LoadContent();
+    public override void LoadContent()
+    {
+        var contentLoader = ContentLoaderFactory.CreateContentLoader();
+
+        // TODO: Need to add ability to change font to the text box.  This means being able to reload
+        // the font when the font type or size changes.
+        this.font = contentLoader.LoadFont("TimesNewRoman-Regular", 12);
```

Comment:
Do you still need this TODO?
```suggestion
        // TODO: Need to add ability to change font to the text box.  This means being able to reload
        // the font when the font type or size changes.
        this.font = contentLoader.LoadFont("TimesNewRoman-Regular", 12);
```